src/primitives/styled.rs

```diff
@@ -110,8 +111,11 @@ impl<T: StyledDimensions<S>, S> Dimensions for Styled<T, S> {
     }
 }
 
-impl<T: StyledDrawable<S>, S> Drawable for Styled<T, S> {
-    type Color = T::Color;
+impl<T: StyledDrawable<S>, S: ColorType> ColorType for Styled<T, S> {
```

Comment:
The `ColorType` trait was the change I wasn't sure about. It was inspired by the `ErrorType` trait in `embedded-hal`. I like the idea of having one common place to specify the color type, instead of associated color types in 6 separate traits, but it causes a lot of changes.

> There are breaking changes, and there are soul-breaking changes, and this feels the latter.

That might be a little dramatic. Would this affect the average user? And most library authors should only need to change a few lines of code.

Reply:
> That might be a little dramatic.

Of course, that came after an hour and a half of reading the PR :)

Reply:
> Of course, that came after an hour and a half of reading the PR :)

OK, I can understand that and thanks for taking the time to review this PR, which should never have gotten so huge.